File: Searching/LinearSearch.cpp
#include <iostream>
using namespace std;

int main()
{
    int arr[] = {1, 2, 45, 3, 5, 6};
    int size = sizeof(arr) / sizeof(arr[0]);
    int target;
    cout << "Enter target value : ";
    cin >> target;

    // Call function to find the value
    int find(int arr[], int size, int target);
    int ans = find(arr, size, target);
    if (ans == -1)
    {
        cout << "Value not found !!";
    }
    else
    {
        cout << "Value present at : " << ans;
    }
    return 0;
}

// Function to find the value
int find(int arr[], int size, int target)
{
    for (int i = 0; i < size; i++)
    {
        if (target == arr[i])
        {
            return i;
        }
    }

    // In case value not found
    return -1;
}

File: Searching/Java Programs/LinearSearch.java
import java.util.*;

public class LinearSearch {
    public static void main(String[] args) throws Exception {

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the size : ");
        int size = sc.nextInt();
        int[] arr = new int[size];

        System.out.println("Enter the values : ");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }

        System.out.print("Enter the target value : ");
        int target = sc.nextInt();

        // Call function to find the value
        int ans = find(arr, target);
        if (ans == -1) {
            System.out.println("Valut not found");
        } else {
            System.out.println("Value present at : " + ans);
        }

    }

    static int find(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i;
            }
        }
        // In case value not found
        return -1;
    }
}
